packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -21,6 +21,20 @@ import 'icon_theme_data.dart';
 /// Typically this is introduced automatically by the [WidgetsApp] or
 /// [MaterialApp].
 ///
+/// {@tool sample}
+///
+/// This example shows how to use [Icon] to create a simple icon in your flutter
```

Comment:
Also: please remove "in your flutter code", it's redundant. See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose

However, maybe you can describe with a word or two what kind of icon the user can expect from this example code.